{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import random\n",
    "from random import shuffle\n",
    "import os\n",
    "import gensim\n",
    "import re\n",
    "import pickle\n",
    "import tensorflow as tf\n",
    "import operator\n",
    "import math\n",
    "import sys\n",
    "from copy import deepcopy\n",
    "from collections import Counter\n",
    "import string\n",
    "import fileinput\n",
    "from nltk.tokenize import sent_tokenize, word_tokenize \n",
    "import nltk\n",
    "\n",
    "from pyspark import SparkContext, SparkConf\n",
    "from pyspark.sql import SQLContext\n",
    "from pyspark.sql.functions import split, explode\n",
    "from pyspark.sql import functions as func\n",
    "from pyspark.sql.types import *\n",
    "from pyspark.ml.linalg import Vectors, DenseVector, VectorUDT\n",
    "\n",
    "from tensorflow.python.ops import rnn, rnn_cell\n",
    "from tensorflow.python.framework import ops\n",
    "from tensorflow.models.rnn.translate import seq2seq_model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "data_dir = '/media/ai2-rey/data_disk/data_sets/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with open(data_dir+'abstracts.txt', 'r') as f:\n",
    "    abstracts = [line.strip() for line in f]\n",
    "    abstracts = abstracts[:1000000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(data_dir+'titles.txt','r') as f:\n",
    "    titles = [line.strip() for line in f]\n",
    "    titles = titles[:1000000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "l = len(abstracts)\n",
    "train_abstracts = abstracts[:int(0.9*l)]\n",
    "train_titles = titles[:int(0.9*l)]\n",
    "\n",
    "test_abstracts = abstracts[int(0.9*l):]\n",
    "test_titles = titles[int(0.9*l):]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "del abstracts\n",
    "del titles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "UNKNOWN = \"*UNKNOWN*\"\n",
    "PAD = \"*PAD*\"\n",
    "\n",
    "def generate_embedding_files(filename):\n",
    "    embeddings = {}\n",
    "    for line in open(filename):\n",
    "        parts = line.split()\n",
    "        embeddings[parts[0]] = list(map(float, parts[1:]))\n",
    "    embedding_size = len(list(embeddings.values())[0])\n",
    "    embeddings[UNKNOWN] = [0.0 for _ in range(embedding_size)]\n",
    "    embeddings[PAD] = [1.0 for _ in range(embedding_size)]\n",
    "    \n",
    "    words = embeddings.keys()\n",
    "    embedding_matrix = np.array([embeddings[word] for word in list(embeddings.keys())])\n",
    "    return words, embedding_matrix\n",
    "\n",
    "glove_words, glove_embedding_matrix = generate_embedding_files('/media/ai2-rey/data_disk/data_sets/glove.6B/glove.6B.50d.txt')\n",
    "glove_vocab_lookup = {word: i for i, word  in enumerate(glove_words)}\n",
    "glove_vocab_size, glove_embedding_size= glove_embedding_matrix.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "class TextSum_Dataset:\n",
    "  \n",
    "    def __init__(self, abstract, title, word2id_dict=None, \n",
    "                 id2word_dict=None, use_glove = True, \n",
    "                 max_vocab_size=None, autopad_abstract='max', \n",
    "                 autopad_title ='max', abstract_pad_len=None, title_pad_len = None):\n",
    "\n",
    "        assert autopad_abstract in {'min', 'max', 'avg'}\n",
    "        assert autopad_title in {'min', 'max', 'avg'}\n",
    "     \n",
    "        self.abstract = abstract\n",
    "        self.title = title\n",
    "        \n",
    "        self.max_vocab_size = max_vocab_size\n",
    "        \n",
    "        self.abstract_pad_len = abstract_pad_len\n",
    "        self.title_pad_len = title_pad_len\n",
    "        self.__autopad_abstract = autopad_abstract\n",
    "        self.__autopad_title = autopad_title\n",
    "        \n",
    "        self.vocab = set()\n",
    "        self.word_counter = dict()\n",
    "        self.use_glove = use_glove\n",
    "        self.word2id_dict = word2id_dict\n",
    "        self.id2word_dict = id2word_dict\n",
    "        \n",
    "        self.abstract_ids = None\n",
    "        self.title_ids = None\n",
    "        self.abstract_lens = None\n",
    "        self.titles_lens = None\n",
    "        \n",
    "        self.__parse_data()\n",
    "        if self.word2id_dict is None: self.__create_word2id_dict() \n",
    "        self.__numericize_data()\n",
    "\n",
    "    def __normalize_text(self, s):\n",
    "        def remove_articles(text):\n",
    "            return re.sub(r'\\b(a|an|the)\\b', ' ', text)\n",
    "\n",
    "        def white_space_fix(text):\n",
    "            return ' '.join(text.split())\n",
    "\n",
    "        def remove_punc(text):\n",
    "            exclude = set(string.punctuation)\n",
    "            exclude.discard('.')\n",
    "            return ''.join(ch for ch in text if ch not in exclude)\n",
    "\n",
    "        def lower(text):\n",
    "            return text.lower()\n",
    "\n",
    "        def add_space(text):\n",
    "            return text.replace('.', ' . ')\n",
    "\n",
    "        return white_space_fix(add_space(remove_punc(lower(s))))\n",
    "\n",
    "    \n",
    "    def __update_word_counter(self, sequence):\n",
    "        \"\"\" Update word_counter with counts for words in a sentence\n",
    "        \n",
    "        Args:\n",
    "            sequence (list<str>) : list of words in a sequence\n",
    "        \n",
    "        \"\"\"\n",
    "        for word in sequence:\n",
    "            self.word_counter[word] = self.word_counter.get(word, 0) + 1\n",
    "            \n",
    "    def __create_vocab(self):\n",
    "        \"\"\" Create set of most frequent unique words found in the training data \"\"\"\n",
    "        \n",
    "        if self.max_vocab_size == None:\n",
    "            self.vocab = set(self.word_counter.keys())\n",
    "        else:\n",
    "            self.vocab = set(sorted(self.word_counter, key=self.word_counter.get, reverse=True)[:self.max_vocab_size])\n",
    "        \n",
    "    def __shuffle_data(self, data):\n",
    "        random.shuffle(data)\n",
    "        return list(zip(*data))\n",
    "    \n",
    "    def __parse_data(self):\n",
    "        \n",
    "        abstracts = []\n",
    "        titles = []\n",
    "        \n",
    "        for idx in range(len(self.abstract)):\n",
    "            a = self.__normalize_text(self.abstract[idx])\n",
    "            t = self.__normalize_text(self.title[idx])\n",
    "            \n",
    "            self.__update_word_counter(a.split())\n",
    "            self.__update_word_counter(t.split())\n",
    "            \n",
    "            abstracts.append(a)\n",
    "            titles.append(t)\n",
    "            \n",
    "        self.abstract = abstracts\n",
    "        self.title = titles\n",
    "        \n",
    "        del abstracts\n",
    "        del titles\n",
    "        \n",
    "        self.__create_vocab()\n",
    "        \n",
    "        shuffle = list(zip(self.abstract, self.title))        \n",
    "        self.abstract, self.title = self.__shuffle_data(shuffle)\n",
    "    \n",
    "    def __create_word2id_dict(self):\n",
    "        \n",
    "        if self.word2id_dict == None:\n",
    "            self.word2id_dict = dict()\n",
    "            self.id2word_dict = dict()\n",
    "            misc_tokens = ['PAD', 'UNK']\n",
    "\n",
    "            for i, token in enumerate(misc_tokens):\n",
    "                self.word2id_dict[token] = i\n",
    "\n",
    "            for word in self.vocab:\n",
    "                self.word2id_dict[word] = len(self.word2id_dict)\n",
    "\n",
    "            self.vocab |= set(misc_tokens)\n",
    "            \n",
    "            word2id = glove_vocab_lookup if self.use_glove else self.word2id_dict\n",
    "            self.id2word_dict = dict(zip(word2id.values(), word2id.keys()))\n",
    "            \n",
    "            self.num_tokens = len(self.word2id_dict)\n",
    "    \n",
    "    def __convert_word2id(self, word):\n",
    "        \n",
    "        try:\n",
    "            word_id = glove_vocab_lookup[word] if self.use_glove else self.word2id_dict[word]\n",
    "        except:\n",
    "            word_id = glove_vocab_lookup[UNKNOWN] if self.use_glove else self.word2id_dict['UNK']\n",
    "\n",
    "        return word_id\n",
    "    \n",
    "    def __apply_padding(self, s, pad_len):\n",
    "        sequence = s[:]\n",
    "        \n",
    "        PAD_sym = glove_vocab_lookup[PAD] if self.use_glove else self.word2id_dict['PAD'] \n",
    "        \n",
    "        if len(sequence) < pad_len:\n",
    "            sequence += [PAD_sym for i in range(pad_len - len(sequence))]\n",
    "        elif len(sequence) > pad_len:\n",
    "            sequence = sequence[:pad_len]\n",
    "        else:\n",
    "            pass\n",
    "        return sequence\n",
    "        \n",
    "    def __get_seq_length_stats(self, sequences):\n",
    "        max_len = 0\n",
    "        min_len = 100000\n",
    "        avg_len = 0\n",
    "        for sequence in sequences:\n",
    "            max_len = max(max_len, len(sequence))\n",
    "            min_len = min(min_len, len(sequence))\n",
    "            avg_len += len(sequence)\n",
    "        avg_len = int(float(avg_len) / len(sequences))\n",
    "        return min_len, max_len, avg_len\n",
    "\n",
    "    def __get_max_sequence_lengths(self, abstract_ids, title_ids):\n",
    "        min_abstract_len, max_abstract_len, avg_abstract_len = self.__get_seq_length_stats(abstract_ids)\n",
    "        min_title_len, max_title_len, avg_title_len = self.__get_seq_length_stats(title_ids)\n",
    "\n",
    "        if self.abstract_pad_len == None:\n",
    "            if self.__autopad_abstract != None:\n",
    "                if self.__autopad_abstract == 'min':\n",
    "                    self.abstract_pad_len = min_abstract_len\n",
    "                elif self.__autopad_abstract == 'max':\n",
    "                    self.abstract_pad_len = max_abstract_len\n",
    "                elif self.__autopad_abstract == 'avg':\n",
    "                    self.abstract_pad_len = avg_abstract_len\n",
    "            else:\n",
    "                self.abstract_pad_len = avg_abstract_len\n",
    "                \n",
    "        if self.title_pad_len == None:\n",
    "            if self.__autopad_title != None:\n",
    "                if self.__autopad_title == 'min':\n",
    "                    self.title_pad_len = min_title_len\n",
    "                elif self.__autopad_title == 'max':\n",
    "                    self.title_pad_len = max_title_len\n",
    "                elif self.__autopad_title == 'avg':\n",
    "                    self.title_pad_len = avg_title_len\n",
    "            else:\n",
    "                self.title_pad_len = avg_title_len     \n",
    "    \n",
    "    def __tokenize_sentences(self, abstract_ids, title_ids):\n",
    "        \"\"\" Tokenizes sentences.\n",
    "        :param raw: dict returned from load_babi\n",
    "        :param word_table: WordTable\n",
    "        :return:\n",
    "        \"\"\"\n",
    "        abstract_tokens = []\n",
    "        title_tokens = []\n",
    "        \n",
    "        for i in range(len(abstract_ids)):\n",
    "            a_tkn_ids = self.__apply_padding(abstract_ids[i], self.abstract_pad_len)\n",
    "            abstract_tokens.append(a_tkn_ids)\n",
    "            t_tkn_ids = self.__apply_padding(title_ids[i], self.title_pad_len)\n",
    "            title_tokens.append(t_tkn_ids)\n",
    "            \n",
    "        return abstract_tokens, title_tokens\n",
    "        \n",
    "    def __convert_text2ids(self, abstract, title):\n",
    "        \"\"\" Tokenizes sentences.\n",
    "        :param raw: dict returned from load_babi\n",
    "        :param word_table: WordTable\n",
    "        :return:\n",
    "        \"\"\"\n",
    "        abstract_ids = []\n",
    "        title_ids = []\n",
    "        \n",
    "        for i in range(len(abstract)):\n",
    "            a_ids = [self.__convert_word2id(word) for word in abstract[i].split()]\n",
    "            abstract_ids.append(a_ids)\n",
    "            t_ids = [self.__convert_word2id(word) for word in title[i].split()]\n",
    "            title_ids.append(t_ids) \n",
    "        \n",
    "        return abstract_ids, title_ids\n",
    "          \n",
    "    \n",
    "    def __numericize_data(self):\n",
    "        a, t = self.__convert_text2ids(self.abstract, self.title)\n",
    "        \n",
    "        self.abstract_lens = [len(i) for i in a]\n",
    "        self.titles_lens = [len(i) for i in t]\n",
    "        \n",
    "        self.__get_max_sequence_lengths(a,t)\n",
    "        \n",
    "        self.abstract_ids, self.title_ids = self.__tokenize_sentences(a,t)\n",
    "        del a,t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "train = TextSum_Dataset(train_abstracts, train_titles, max_vocab_size=50000, abstract_pad_len=200, title_pad_len=20, use_glove=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test = TextSum_Dataset(test_abstracts, test_titles, train.word2id_dict, train.id2word_dict, \n",
    "                      abstract_pad_len = train.abstract_pad_len, title_pad_len = train.title_pad_len, use_glove=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50002"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(train.word2id_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "del train_abstracts, train_titles, test_abstracts, test_titles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "class DataIterator:  \n",
    "    def __init__(self, data, batch_size):\n",
    "        self.data = data\n",
    "        self.batch_size = batch_size\n",
    "        self.data_iterator = self.make_random_iter()\n",
    "        \n",
    "    def next_batch(self):\n",
    "        try:\n",
    "            idxs = next(self.data_iterator)\n",
    "        except StopIteration:\n",
    "            self.data_iterator = self.make_random_iter()\n",
    "            idxs = next(self.data_iterator)\n",
    "            \n",
    "        batch = [self.data[i] for i in idxs]\n",
    "        batch_idxs = [idx for idx in idxs]\n",
    "        return batch, batch_idxs\n",
    "\n",
    "    def make_random_iter(self):\n",
    "        splits = np.arange(self.batch_size, len(self.data), self.batch_size)\n",
    "        it = np.split(np.random.permutation(range(len(self.data))), splits)[:-1]\n",
    "        return iter(it)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_ids = list(zip(train.abstract_ids, train.title_ids))\n",
    "test_ids = list(zip(test.abstract_ids, test.title_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_data_iter = DataIterator(train_ids,128)\n",
    "test_data_iter = DataIterator(test_ids,128)\n",
    "deploy_data_iter = DataIterator(train_ids,1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def prepare_model_dir():\n",
    "    model_dir = os.getcwd() + '/'\n",
    "    if not os.path.exists(model_dir + 'weights'):\n",
    "        os.makedirs(model_dir + 'weights')\n",
    "    return model_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def write_to_log(string,filename):\n",
    "#     print(string)\n",
    "    with open(filename,'a') as write_file:\n",
    "        write_file.write(string + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def loop_function(prev, embedding, w, b, update_embedding=True):\n",
    "    prev = tf.nn.xw_plus_b(prev, w, b)\n",
    "    prev_symbol = tf.argmax(prev,1) #maybe 0 if not transpose\n",
    "    emb_prev = tf.nn.embedding_lookup(embedding, prev_symbol)\n",
    "    if not update_embedding:\n",
    "        emb_prev = tf.stop_gradient(emb_prev)\n",
    "    return "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "abstract_max_len = train.abstract_pad_len\n",
    "title_max_len = train.title_pad_len\n",
    "\n",
    "n_hidden = 150 \n",
    "n_hidden_layers = 2\n",
    "word_dim = glove_embedding_size\n",
    "vocab_size = len(train.vocab)\n",
    "\n",
    "learning_rate = 0.001\n",
    "train_iters = 100000\n",
    "keep_prob = 0.5\n",
    "\n",
    "display_step = 200\n",
    "val_interval = 1000\n",
    "save_weights_interval = 1000\n",
    "deploy_interval = 1000\n",
    "\n",
    "\n",
    "device = '/gpu:0'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "model_dir = prepare_model_dir()\n",
    "tf.reset_default_graph()\n",
    "sess = tf.InteractiveSession()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "passage = tf.placeholder(tf.int32,[None, abstract_max_len])\n",
    "#passage_lens = tf.placeholder(tf.int32,[None])\n",
    "summary = tf.placeholder(tf.int64,[None, title_max_len])\n",
    "\n",
    "dropout = tf.placeholder(tf.float32)\n",
    "batch_size = tf.placeholder(tf.int32)\n",
    "use_prev = tf.placeholder(tf.int32)\n",
    "\n",
    "encoder_inputs = tf.unstack(passage, axis=1)\n",
    "decoder_inputs = tf.unstack(summary, axis=1)\n",
    "decoder_inputs = [tf.zeros_like(decoder_inputs[0], dtype=tf.int64, name='GO')] + decoder_inputs[:-1]\n",
    "\n",
    "with tf.variable_scope('embedding'):\n",
    "    embedding = tf.Variable(tf.truncated_normal(shape=[vocab_size, word_dim], stddev=1e-4))\n",
    "    #embedding = tf.constant(glove_embedding_matrix, name='embeddings', dtype=tf.float32)\n",
    "    \n",
    "    emb_enc_inputs = [tf.nn.embedding_lookup(embedding, x)\n",
    "                      for x in encoder_inputs]\n",
    "    emb_dec_inputs = [tf.nn.embedding_lookup(embedding, x)\n",
    "                      for x in decoder_inputs]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with tf.variable_scope('encoder'):\n",
    "    enc_cell = tf.nn.rnn_cell.LSTMCell(n_hidden)\n",
    "    enc_cell = tf.nn.rnn_cell.DropoutWrapper(enc_cell, output_keep_prob=dropout)\n",
    "    #enc_cell = tf.nn.rnn_cell.MultiRNNCell([enc_cell]*n_hidden_layers)\n",
    "\n",
    "    enc_state = enc_cell.zero_state(batch_size, dtype=tf.float32)\n",
    "    h_states = []\n",
    "    c_states = []\n",
    "    \n",
    "    for i in range(abstract_max_len):\n",
    "        h, enc_state = enc_cell(emb_enc_inputs[i], enc_state)\n",
    "        tf.get_variable_scope().reuse_variables()\n",
    "#         h_states.append(enc_state.h)\n",
    "#         c_states.append(enc_state.c)\n",
    "    \n",
    "#     h_states = tf.transpose(tf.stack(h_states),[1,0,2])\n",
    "#     h_indices = tf.range(tf.shape(h_states)[0])*tf.shape(h_states)[1]+(passage_lens-1)\n",
    "#     h_thought = tf.gather(tf.reshape(h_states,[-1,n_hidden]), h_indices)\n",
    "    \n",
    "#     c_states = tf.transpose(tf.stack(c_states),[1,0,2])\n",
    "#     c_indices = tf.range(tf.shape(c_states)[0])*tf.shape(c_states)[1]+(passage_lens-1)\n",
    "#     c_thought = tf.gather(tf.reshape(c_states,[-1,n_hidden]), c_indices)\n",
    "\n",
    "w = tf.Variable(tf.truncated_normal(shape=[n_hidden, vocab_size], stddev=1e-4))\n",
    "w_t = tf.transpose(w)\n",
    "b = tf.Variable(tf.truncated_normal(shape=[vocab_size], stddev=1e-4))\n",
    "\n",
    "with tf.variable_scope('decoder'):\n",
    "    \n",
    "    dec_cell = tf.nn.rnn_cell.LSTMCell(n_hidden)\n",
    "    dec_cell = tf.nn.rnn_cell.DropoutWrapper(dec_cell, output_keep_prob=dropout)\n",
    "    dec_cell = tf.nn.rnn_cell.OutputProjectionWrapper(dec_cell, vocab_size)\n",
    "    \n",
    "    dec_input_state =  enc_state\n",
    "    dec_outputs, dec_state = tf.nn.seq2seq.rnn_decoder(emb_dec_inputs, dec_input_state,\n",
    "                                               dec_cell)\n",
    "\n",
    "with tf.variable_scope('loss'):\n",
    "    weights = [tf.ones_like(label,dtype=tf.float32) for label in decoder_inputs]\n",
    "    loss = tf.nn.seq2seq.sequence_loss(dec_outputs, decoder_inputs, weights, vocab_size)\n",
    "\n",
    "with tf.variable_scope('prediction'):\n",
    "    accuracy = 0\n",
    "    generated_tokens = []\n",
    "    for i,output in enumerate(dec_outputs):\n",
    "        output = tf.nn.softmax(output)\n",
    "        pred_tokens = tf.argmax(output, axis=1)\n",
    "        generated_tokens.append(pred_tokens)\n",
    "        \n",
    "        label = tf.slice(summary,[0,i],[-1,1])\n",
    "        accuracy += tf.reduce_mean(tf.cast(tf.equal(pred_tokens,label), tf.float32))\n",
    "    accuracy = accuracy / float(title_max_len)\n",
    "    \n",
    "optimizer = tf.train.AdamOptimizer(learning_rate=learning_rate).minimize(loss)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sess.run(tf.global_variables_initializer())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "log_file = model_dir + 'seq2seq_log.txt'\n",
    "saver = tf.train.Saver(max_to_keep=100)\n",
    "log=open(log_file,'w')\n",
    "log.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Iter 1, accuracy = 0.0001, loss = 10.819901\n",
      "Iter 200, accuracy = 0.1541, loss = 4.206182\n",
      "Iter 400, accuracy = 0.1591, loss = 3.578858\n",
      "Iter 600, accuracy = 0.1736, loss = 2.922408\n",
      "Iter 800, accuracy = 0.1725, loss = 2.260313\n",
      "Iter 1000, accuracy = 0.1558, loss = 1.934989\n",
      "Iter 1000, VALIDATION -- accuracy = 0.1785, loss = 1.944571\n",
      "PASSAGE\n",
      "22q11 . 2 deletion syndrome 22q11 . 2ds is a multisystemic disorder with high phenotypic variability . underdiagnosis in adults is common and recognition of facial dysmorphic features can be affected by age and ethnicity . this study aims to determine the prevalence of undiagnosed 22q11 . 2ds in adult chinese patients with UNK anomalies and to delineate their facial UNK and extracardiac manifestations . we recruited consecutively 156 patients with UNK anomalies in an adult congenital heart disease chd clinic in hong kong and screened for 22q11 . 2ds using UNK and fluorescence insitu hybridization . assessment for dysmorphic features was performed by a cardiologist at initial screening and then by a clinical UNK upon result disclosure . clinical photographs were taken and childhood photographs collected . eighteen patients 11 . 5 were diagnosed with 22q11 . 2ds translating into 1 previously unrecognized diagnosis of 22q11 . 2ds in every 10 adult patients with UNK anomalies . while dysmorphic features were detected by our clinical UNK in all patients only twothirds were considered dysmorphic by our cardiologist upon first assessment . evolution of facial dysmorphic features was noted with age . extracardiac manifestations included velopharyngeal incompetence or cleft palate\n",
      "Model Pred = PAD UNK of bacterial . 2 temperature in adult chinese patients with UNK weight implications in extraction care . PAD\n",
      "Ground Truth = UNK of 22q11 . 2 deletion in adult chinese patients with UNK anomalies implications in transitional care . PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--1000.ckpt\n",
      "Iter 1200, accuracy = 0.1930, loss = 1.609255\n",
      "Iter 1400, accuracy = 0.1549, loss = 1.488688\n",
      "Iter 1600, accuracy = 0.2052, loss = 1.180087\n",
      "Iter 1800, accuracy = 0.1729, loss = 1.072663\n",
      "Iter 2000, accuracy = 0.1668, loss = 0.962659\n",
      "Iter 2000, VALIDATION -- accuracy = 0.1897, loss = 0.912719\n",
      "PASSAGE\n",
      "free fatty acids ffas are dietary nutrients which act as ligands for ffas receptors . gproteincoupled receptor 120 gpr120 and gpr40 are activated by long and medium chain ffas . in the present study we investigated the role of the gpr120 and gpr40 in cell motile activity stimulated by UNK in rat liver epithelial UNK cells . cells were treated with UNK at a concentration of 10μm every 24h for UNK . the expression levels of the gpr120 and gpr40 genes in UNK cells treated UNK were significantly higher than those in untreated cells . in cell motility assay the cell motile activity of UNK cells was markedly elevated by UNK compared with untreated cells . to evaluate the effects of gpr120 on the cell motile activity by UNK we established gpr120 knockdown cells from UNK cells . the cell motile activity stimulated by UNK was significantly suppressed by gpr120 knockdown . in addition a potent gpr40 antagonist UNK enhanced the cell motile activity by UNK . these results suggest that opposite effects of gpr120 and gpr40 may be involved in the cell motile activity stimulated by UNK in UNK cells . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD PAD effects of PAD and saudi on cell PAD activity induced by UNK in liver epithelial cells . PAD\n",
      "Ground Truth = opposite effects of gpr120 and gpr40 on cell motile activity induced by UNK in liver epithelial cells . PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--2000.ckpt\n",
      "Iter 2200, accuracy = 0.1521, loss = 0.797600\n",
      "Iter 2400, accuracy = 0.1598, loss = 0.847316\n",
      "Iter 2600, accuracy = 0.1688, loss = 0.623804\n",
      "Iter 2800, accuracy = 0.1649, loss = 0.655960\n",
      "Iter 3000, accuracy = 0.1767, loss = 0.556493\n",
      "Iter 3000, VALIDATION -- accuracy = 0.1589, loss = 0.529139\n",
      "PASSAGE\n",
      "UNK UNK ammonium chitosan chloride UNK is UNK chitosan cs derivative which can be obtained by the reaction between UNK trimethyl ammonium chloride eta and cs . the preparation parameters for the synthesis of UNK were optimized by orthogonal experimental design . eta was successfully grafted into the free amino group of cs . grafting of eta with cs had great effect on the crystal structure of UNK which was confirmed by the xrd results . UNK displayed higher capability to form nanoparticles by crosslinking with negatively charged sodium UNK tpp . UNK UNK loaded UNK nanoparticles were positively charged and were spherical in shape with average particle size of 200 nm . more efficient drug encapsulation efficiency and loading capacity were obtained for UNK in comparison with cs however UNK nanoparticles displayed faster release rate due to its UNK properties . the results suggest that UNK is a promising cs derivative for the encapsulation of hydrophilic drugs in obtaining sustained release of drugs . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD synthesis and characterization of chitosan quaternary ammonium salt and its application as drug carrier for ribavirin . PAD PAD\n",
      "Ground Truth = synthesis and characterization of chitosan quaternary ammonium salt and its application as drug carrier for ribavirin . PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--3000.ckpt\n",
      "Iter 3200, accuracy = 0.1667, loss = 0.503617\n",
      "Iter 3400, accuracy = 0.1595, loss = 0.498573\n",
      "Iter 3600, accuracy = 0.1660, loss = 0.389721\n",
      "Iter 3800, accuracy = 0.1990, loss = 0.395470\n",
      "Iter 4000, accuracy = 0.1869, loss = 0.415331\n",
      "Iter 4000, VALIDATION -- accuracy = 0.1761, loss = 0.301777\n",
      "PASSAGE\n",
      "UNK disease is caused by an inherited deficiency of the enzyme UNK resulting in exercise intolerance from childhood and acute crises of early fatigue and contractures . in severe cases these manifestations can be accompanied by rhabdomyolysis UNK and fatal renal failure . diagnosis of UNK disease is based on clinical diagnostic tests together with an absence of UNK activity in skeletal muscle biopsies and genetic analysis of the UNK gene UNK . the recently reported association between UNK and rac1 gtpase in a t lymphocyte cell line prompted us to study UNK expression in white blood cells wbcs from 20 healthy donors and 30 UNK patients by flow cytometry using a UNK UNK antibody . we found that t lymphocytes expressed UNK in healthy donors but expression was significantly lower in UNK patients p0 . 001 . UNK mrna levels were also lower in white blood cells from UNK patients . nevertheless in 13 of patients who were either heterozygotes or homozygotes for the most common UNK pathogenic mutation among caucasians p . UNK the percentage of UNK white blood cells was not different compared with the control group . our findings suggest that analysis of UNK expression in\n",
      "Model Pred = PAD UNK expression analysis in white blood cells a complementary tool for diagnosing UNK disease PAD PAD PAD PAD PAD\n",
      "Ground Truth = UNK expression analysis in white blood cells a complementary tool for diagnosing UNK disease PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--4000.ckpt\n",
      "Iter 4200, accuracy = 0.1720, loss = 0.331873\n",
      "Iter 4400, accuracy = 0.1782, loss = 0.286025\n",
      "Iter 4600, accuracy = 0.1322, loss = 0.292144\n",
      "Iter 4800, accuracy = 0.1916, loss = 0.248987\n",
      "Iter 5000, accuracy = 0.1687, loss = 0.211686\n",
      "Iter 5000, VALIDATION -- accuracy = 0.2076, loss = 0.232364\n",
      "PASSAGE\n",
      "development of new fruit cultivars is dependent on genetic resource collections such as those at the UNK garden of the institute of UNK and UNK near UNK kazakhstan . the pear germplasm collection of the UNK garden contains 615 cultivars and three species . in vitro cold storage of the collection would provide additional security to the field collection . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD growth medium alterations improve in vitro cold storage of pear germplasm . PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = growth medium alterations improve in vitro cold storage of pear germplasm . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--5000.ckpt\n",
      "Iter 5200, accuracy = 0.1703, loss = 0.258094\n",
      "Iter 5400, accuracy = 0.1803, loss = 0.201676\n",
      "Iter 5600, accuracy = 0.1641, loss = 0.177796\n",
      "Iter 5800, accuracy = 0.1597, loss = 0.209063\n",
      "Iter 6000, accuracy = 0.1441, loss = 0.175756\n",
      "Iter 6000, VALIDATION -- accuracy = 0.1881, loss = 0.167548\n",
      "PASSAGE\n",
      "in mammals the auditory system which includes the cochlea has a very complex structure harboring many types of cells performing different functions . among these cells are the auditory hair cells hcs which are terminally and well differentiated unique cells which have lost their regenerative potential after development . the auditory hcs are easily damaged by aging as well as during episodes of ototoxicity and acoustic trauma . hcs damages typically occur in the early stage of injury and can result a permanent hearing loss . recently there have been tremendous developments from stem cells scs research involving sensorineural hearing loss but several limitations and obstacles persist in allowing these developments from continuing onto clinical applications . this review discusses the recent advances in sc research in sensorineural hearing loss with the subsequent sections discussing the possible hurdles and limitations that currently preclude their clinical application . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD stem cell therapy for sensorineural hearing loss still PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = stem cell therapy for sensorineural hearing loss still alive PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--6000.ckpt\n",
      "Iter 6200, accuracy = 0.1557, loss = 0.190039\n",
      "Iter 6400, accuracy = 0.1913, loss = 0.163722\n",
      "Iter 6600, accuracy = 0.1587, loss = 0.110151\n",
      "Iter 6800, accuracy = 0.1896, loss = 0.155638\n",
      "Iter 7000, accuracy = 0.1565, loss = 0.134220\n",
      "Iter 7000, VALIDATION -- accuracy = 0.1653, loss = 0.107794\n",
      "PASSAGE\n",
      "although bone involvement associated with cyst hydatid is rarely seen it can cause unintended results such as high recurrence rate infection sepsis or amputation of relevant extremity . because of this reason its treatment is difficult and disputed . in the case of UNK bone cyst hydatid in right femur along with albendazole treatment result of UNK cyst surgically and its treatment with teicoplanin with added bone cement is given . in conclusion since the offered treatment method both supports bone in terms of mechanical aspect and also can prevent secondary infection the method is thought to be a good and safe treatment approach . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD treatment of UNK cyst hydatid involvement in right femur with teicoplanin added bone cement and albendazole . PAD PAD\n",
      "Ground Truth = treatment of UNK cyst hydatid involvement in right femur with teicoplanin added bone cement and albendazole . PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--7000.ckpt\n",
      "Iter 7200, accuracy = 0.1664, loss = 0.088933\n",
      "Iter 7400, accuracy = 0.1830, loss = 0.111431\n",
      "Iter 7600, accuracy = 0.1799, loss = 0.071980\n",
      "Iter 7800, accuracy = 0.1457, loss = 0.073122\n",
      "Iter 8000, accuracy = 0.1636, loss = 0.065225\n",
      "Iter 8000, VALIDATION -- accuracy = 0.1488, loss = 0.085947\n",
      "PASSAGE\n",
      "green tea and hibiscus are widely consumed as traditional beverages in yemen and some regional countries . they are relatively cheap and the belief is that they improve health state and cure many diseases . the aim of this study was to evaluate the potential protective and antibacterial activity of these two famous plants in vitro through measuring their antibacterial activity and in vivo through measuring nonenzymatic kidney markers dysfunction after induction of nephrotoxicity by gentamicin . gram positive bacteria like mrsa methicillin resistant staphylococcus aureus were isolated from hospitalized patients different sources pus and wound and gram negative bacteria including e . coli and p . aeruginosa were used in vitro study . in addition the efficacy of these plants was assessed in vivo through measuring nonenzymatic kidney markers including s . creatinine and s . urea . green tea was shown antimicrobial activity against mrsa with inhibition zone 19 . 67 ± 0 . 33 mm and mic 1 . 25 ± 0 . 00 mgml compared with standard reference vancomycin 18 . 00 ± 0 . 00 mgml . hibiscus did not exhibit a similar effect . both hibiscus and green UNK groups had nephroprotective effects as\n",
      "Model Pred = PAD evaluation of the potential nephroprotective and antimicrobial effect of camellia sinensis leaves versus hibiscus UNK in vivo and in\n",
      "Ground Truth = evaluation of the potential nephroprotective and antimicrobial effect of camellia sinensis leaves versus hibiscus UNK in vivo and in vitro\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--8000.ckpt\n",
      "Iter 8200, accuracy = 0.1705, loss = 0.066324\n",
      "Iter 8400, accuracy = 0.1653, loss = 0.041354\n",
      "Iter 8600, accuracy = 0.1884, loss = 0.059215\n",
      "Iter 8800, accuracy = 0.1850, loss = 0.063735\n",
      "Iter 9000, accuracy = 0.1975, loss = 0.043090\n",
      "Iter 9000, VALIDATION -- accuracy = 0.1664, loss = 0.094751\n",
      "PASSAGE\n",
      "voltagegated potassium kv channels are involved in action potential ap repolarization in excitable cells . exogenous application of UNK lipids such as arachidonic acid aa regulates the gating of kv channels . whether UNK lipids released under physiological conditions have an impact on neuronal coding through this mechanism is unknown . we show that aa released in an activitydependent manner from postsynaptic hippocampal ca3 pyramidal cells acts as retrograde messenger inducing a robust facilitation of mossy fiber mf synaptic transmission over several minutes . aa acts by broadening presynaptic aps through the direct modulation of kv channels . this form of shortterm plasticity can be triggered when postsynaptic cell fires with physiologically relevant patterns and sets the threshold for the induction of the presynaptic form of longterm potentiation ltp at hippocampal mf synapses . hence direct modulation of presynaptic kv channels by activitydependent release of lipids serves as a physiological mechanism for tuning synaptic transmission . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD membrane lipids tune synaptic transmission by direct modulation of presynaptic potassium channels . PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = membrane lipids tune synaptic transmission by direct modulation of presynaptic potassium channels . PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--9000.ckpt\n",
      "Iter 9200, accuracy = 0.1849, loss = 0.041473\n",
      "Iter 9400, accuracy = 0.1852, loss = 0.052025\n",
      "Iter 9600, accuracy = 0.1765, loss = 0.026402\n",
      "Iter 9800, accuracy = 0.1700, loss = 0.039223\n",
      "Iter 10000, accuracy = 0.1891, loss = 0.050801\n",
      "Iter 10000, VALIDATION -- accuracy = 0.1555, loss = 0.071809\n",
      "PASSAGE\n",
      "UNK is a mainly nuclear catalytic subunit of the human exosome complex containing UNK UNK and UNK pin active domains . mutations in UNK have been found in ∼10 of patients with multiple myeloma mm . here we show that these mutations interfere with UNK UNK activity . yeast harboring corresponding mutations in UNK show growth inhibition and changes in nuclear rna metabolism typical for exosome dysfunction . construction of a conditional UNK knockout in the chicken dt40 cell line revealed that UNK is essential for cell survival indicating that its function cannot be replaced by other UNK nucleases UNK and UNK . moreover UNK cells in which depletion of endogenous wildtype UNK was complemented with exogenously expressed mm UNK mutants proliferate at a slower rate and exhibit aberrant rna metabolism . importantly mm mutations are synthetically lethal with the UNK pin domain catalytic mutation both in yeast and human cells . since mutations in pin domain alone have little effect on cell physiology our results predict the UNK pin domain as a potential drug target for mm patients with UNK mutations . it is an interesting example of intramolecular synthetic lethality with putative therapeutic potential in humans . PAD\n",
      "Model Pred = PAD multiple UNK UNK mutations cause perturbations in cellular rna metabolism and suggest UNK pin domain as a potential drug\n",
      "Ground Truth = multiple UNK UNK mutations cause perturbations in cellular rna metabolism and suggest UNK pin domain as a potential drug target\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--10000.ckpt\n",
      "Iter 10200, accuracy = 0.1676, loss = 0.035023\n",
      "Iter 10400, accuracy = 0.1783, loss = 0.033721\n",
      "Iter 10600, accuracy = 0.1302, loss = 0.042962\n",
      "Iter 10800, accuracy = 0.1622, loss = 0.023711\n",
      "Iter 11000, accuracy = 0.1717, loss = 0.023356\n",
      "Iter 11000, VALIDATION -- accuracy = 0.1679, loss = 0.021319\n",
      "PASSAGE\n",
      "the purpose of this report was to evaluate the role antivegf in management of UNK secondary to ocular toxoplasmosis . young female diagnosed as a case of bilateral ocular toxoplasmosis presented with complaints of diminution of vision in the right eye . fundus examination showed an active UNK adjacent to toxoplasmosis scar . in view of active UNK patient was administered intravitreal ranibizumab . a total of 2 injections of intravitreal ranibizumab were given . fundus showed a scarred UNK adjacent to the toxoplasma scar with no clinical signs of activity . antivegf therapy has been successfully used to improve visual and anatomical outcome in UNK deleted UNK secondary to toxoplasmosis . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD paediatric choroidal neovascular membrane secondary to toxoplasmosis treated successfully with antivascular endothelial growth factor . PAD PAD PAD PAD\n",
      "Ground Truth = paediatric choroidal neovascular membrane secondary to toxoplasmosis treated successfully with antivascular endothelial growth factor . PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--11000.ckpt\n",
      "Iter 11200, accuracy = 0.1505, loss = 0.030320\n",
      "Iter 11400, accuracy = 0.2138, loss = 0.038206\n",
      "Iter 11600, accuracy = 0.1664, loss = 0.017679\n",
      "Iter 11800, accuracy = 0.1420, loss = 0.013074\n",
      "Iter 12000, accuracy = 0.1756, loss = 0.019553\n",
      "Iter 12000, VALIDATION -- accuracy = 0.1373, loss = 0.030170\n",
      "PASSAGE\n",
      "in this study novel UNK ultraviolet lightemitting diodes UNK were fabricated using wafer bonding laser liftoff dry etching textured surface and interconnection techniques . placing pn electrodes on the same side minimized the absorption caused by electrodes in conventional vertical injection lightemitting diodes UNK and the current spreading was improved . the light output power 700 ma of the UNK was enhanced by 18 . 3 compared with that of the UNK and the external quantum efficiency eqe of the UNK was also relatively enhanced by 20 . 0 compared with that of a reference device . when the current operations were 1500 ma the enhancements of the light output power and eqe were 27 . 4 and 27 . 2 respectively . additionally the efficiency UNK was improved by more than 15 at the same current level . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD novel UNK led structure adopted micro UNK surface to compare with conventional vertical leds in ultraviolet light . PAD\n",
      "Ground Truth = novel UNK led structure adopted micro UNK surface to compare with conventional vertical leds in ultraviolet light . PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--12000.ckpt\n",
      "Iter 12200, accuracy = 0.1379, loss = 0.051073\n",
      "Iter 12400, accuracy = 0.1555, loss = 0.028593\n",
      "Iter 12600, accuracy = 0.1737, loss = 0.006492\n",
      "Iter 12800, accuracy = 0.1437, loss = 0.014613\n",
      "Iter 13000, accuracy = 0.1471, loss = 0.012876\n",
      "Iter 13000, VALIDATION -- accuracy = 0.1729, loss = 0.014754\n",
      "PASSAGE\n",
      "the use of intensity modulated radiation therapy imrt has allowed for the administration of high doses to retroperitoneal sarcomas UNK while limiting toxicity to adjacent organs . the purpose of our study is to assess the outcome and toxicities of patients with UNK treated with neoadjuvant external beam radiation ebrt therapy using imrt . this is a retrospective study of 21 patients treated with preoperative imrt for primary or recurrent UNK between 2005 and 2011 . overall survival os and local recurrence free survival UNK were computed using the kaplanmeier method logrank test . acute and chronic toxicities were assessed using the UNK v . 3 criteria . the actuarial 2 and 3year os was 66 for both and the 5year os was 51 . as for UNK it was 57 at 2 and 3year and 51 for the 5year UNK . factors predictive for local control were microscopically negative margins p 0 . 022 a median tumor diameter 15 cm p 0 . 007 and pathology of liposarcoma p 0 . 021 . furthermore patients treated for recurrent disease UNK worse p 0 . 04 in local control than patients treated for primary disease . as for os patients\n",
      "Model Pred = PAD preoperative intensity modulated radiation therapy for retroperitoneal sarcoma . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = preoperative intensity modulated radiation therapy for retroperitoneal sarcoma . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--13000.ckpt\n",
      "Iter 13200, accuracy = 0.1535, loss = 0.016081\n",
      "Iter 13400, accuracy = 0.1865, loss = 0.016792\n",
      "Iter 13600, accuracy = 0.1654, loss = 0.009282\n",
      "Iter 13800, accuracy = 0.1685, loss = 0.006274\n",
      "Iter 14000, accuracy = 0.1684, loss = 0.010537\n",
      "Iter 14000, VALIDATION -- accuracy = 0.1655, loss = 0.031209\n",
      "PASSAGE\n",
      "singlemolecule localization microscopy has been widely applied to count the number of biological molecules within a certain structure . the percentage of molecules that are detected significantly affects the interpretation of data . among many factors that affect this percentage the polarization state of the excitation light is often neglected or at least UNK in publications . we demonstrate by simulation and experiment that the number of molecules detected can be different from 40 up to 100 when using circularly or linearly polarized excitation light . this is determined mainly by the number of photons emitted by single fluorescent molecule namely the choice of fluorescence proteins and the background noise in the system namely the illumination scheme . this difference can be further exaggerated or mitigated by various fixation methods magnification and camera settings we conclude that the final choice between circularly or linearly polarized excitation light should be made experimentally based on the signal to noise ratio of the system . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD polarization of excitation light influences molecule counting in singlemolecule localization microscopy . PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = polarization of excitation light influences molecule counting in singlemolecule localization microscopy . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--14000.ckpt\n",
      "Iter 14200, accuracy = 0.1816, loss = 0.010177\n",
      "Iter 14400, accuracy = 0.1544, loss = 0.009640\n",
      "Iter 14600, accuracy = 0.1558, loss = 0.018028\n",
      "Iter 14800, accuracy = 0.1283, loss = 0.007857\n",
      "Iter 15000, accuracy = 0.1543, loss = 0.006997\n",
      "Iter 15000, VALIDATION -- accuracy = 0.1737, loss = 0.046854\n",
      "PASSAGE\n",
      "recent advances in behavioral genetics suggest a modest relationship among certain gene variants early childhood experiences and criminal behavior . although scientific research examining this link is still at an early stage genetic data are already being introduced in criminal trials . however the extent to which such evidence is likely to affect UNK decisions has not been explored . in the present study a representative sample of the u . s . population n 250 received a vignette describing an apparently impulsive homicide accompanied by one of four explanations of the UNK impulsivity childhood abuse genetic predisposition childhood abuse and genetic predisposition or simple impulsive behavior . the participants were asked to identify the crime that the UNK had committed and to select an appropriate sentence range . evidence of genetic predisposition did not affect the crime of which the UNK was UNK or the sentence . however participants who received the abuse or genetic abuse explanation imposed longer prison sentences . paradoxically the genetic and genetic abuse conditions engendered the greatest fear of the UNK . these findings should UNK concerns that genetic evidence in criminal UNK will be overly persuasive to UNK but should raise questions about\n",
      "Model Pred = PAD impact of behavioral genetic evidence on the adjudication of criminal behavior . PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = impact of behavioral genetic evidence on the adjudication of criminal behavior . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--15000.ckpt\n",
      "Iter 15200, accuracy = 0.1595, loss = 0.002653\n",
      "Iter 15400, accuracy = 0.1657, loss = 0.007393\n",
      "Iter 15600, accuracy = 0.1628, loss = 0.005066\n",
      "Iter 15800, accuracy = 0.1734, loss = 0.015159\n",
      "Iter 16000, accuracy = 0.1719, loss = 0.010448\n",
      "Iter 16000, VALIDATION -- accuracy = 0.1503, loss = 0.016341\n",
      "PASSAGE\n",
      "nausea and vomiting are common symptoms in palliative care and can be highly distressing to patients . this review discusses the mechanisms by which nausea and vomiting are triggered using case studies to highlight the most common scenarios and how to manage these . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD management of nausea and vomiting in palliative care . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = management of nausea and vomiting in palliative care . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--16000.ckpt\n",
      "Iter 16200, accuracy = 0.2149, loss = 0.007601\n",
      "Iter 16400, accuracy = 0.1503, loss = 0.001075\n",
      "Iter 16600, accuracy = 0.1560, loss = 0.022901\n",
      "Iter 16800, accuracy = 0.1733, loss = 0.018037\n",
      "Iter 17000, accuracy = 0.1462, loss = 0.005611\n",
      "Iter 17000, VALIDATION -- accuracy = 0.1774, loss = 0.005489\n",
      "PASSAGE\n",
      "diabetic macular edema dme is a UNK complication of diabetic retinopathy . the current practice of management is a trial and error method of using intravitreal antivascular endothelial growth factor vegf or steroids to treat the patient and watch the response . however if the patients genetic profile helps us choose appropriate medicine it would help UNK treatment option for each patient . this forms the basis of our study . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD differential systemic gene expression profile in patients with diabetic macular edema responders versus nonresponders to standard treatment . PAD\n",
      "Ground Truth = differential systemic gene expression profile in patients with diabetic macular edema responders versus nonresponders to standard treatment . PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--17000.ckpt\n",
      "Iter 17200, accuracy = 0.1773, loss = 0.005433\n",
      "Iter 17400, accuracy = 0.1622, loss = 0.003203\n",
      "Iter 17600, accuracy = 0.1391, loss = 0.004717\n",
      "Iter 17800, accuracy = 0.1590, loss = 0.008225\n",
      "Iter 18000, accuracy = 0.1760, loss = 0.011067\n",
      "Iter 18000, VALIDATION -- accuracy = 0.1714, loss = 0.012854\n",
      "PASSAGE\n",
      "inorganic arsenic is methylated in the body by arsenic iii methyltransferase UNK . arsenic methylation is thought to play a role in UNK epigenetic phenomena including aberrant dna and histone methylation . however it is unclear whether the promoter of the UNK gene which codes for UNK is differentially methylated as a function of arsenic exposure . in this study we evaluated UNK promoter methylation according to exposure assessed by urinary arsenic excretion in a stratified random sample of 48 participants from the strong heart study who had urine arsenic measured at baseline and dna available from 1989 to 1991 and UNK . for this study all data are from the UNK visit . we measured UNK promoter methylation at its 48 cpg loci by UNK sequencing . we compared mean methylation at each cpg locus by arsenic exposure group using linear regression adjusted for study centre age and sex . a hypomethylated region in the UNK promoter was associated with higher arsenic exposure . in vitro arsenic induced UNK promoter hypomethylation and it increased UNK expression in human peripheral blood mononuclear cells . these findings may suggest that arsenic exposure influences the epigenetic regulation of a major arsenic metabolism\n",
      "Model Pred = PAD differential methylation of the arsenic iii methyltransferase promoter according to arsenic exposure . PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = differential methylation of the arsenic iii methyltransferase promoter according to arsenic exposure . PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--18000.ckpt\n",
      "Iter 18200, accuracy = 0.1609, loss = 0.017950\n",
      "Iter 18400, accuracy = 0.1764, loss = 0.010013\n",
      "Iter 18600, accuracy = 0.1599, loss = 0.006788\n",
      "Iter 18800, accuracy = 0.1889, loss = 0.004726\n",
      "Iter 19000, accuracy = 0.1432, loss = 0.005862\n",
      "Iter 19000, VALIDATION -- accuracy = 0.1655, loss = 0.005339\n",
      "PASSAGE\n",
      "this article reviews recent evidence and other earlier relevant articles regarding early intervention studies for children with autism spectrum disorder asd . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD a review of evidencebased early intervention for behavioural problems in children with autism spectrum disorder the core components of\n",
      "Ground Truth = a review of evidencebased early intervention for behavioural problems in children with autism spectrum disorder the core components of effective\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--19000.ckpt\n",
      "Iter 19200, accuracy = 0.1667, loss = 0.004311\n",
      "Iter 19400, accuracy = 0.1983, loss = 0.005238\n",
      "Iter 19600, accuracy = 0.1764, loss = 0.006231\n",
      "Iter 19800, accuracy = 0.1755, loss = 0.003989\n",
      "Iter 20000, accuracy = 0.1659, loss = 0.008567\n",
      "Iter 20000, VALIDATION -- accuracy = 0.1771, loss = 0.006781\n",
      "PASSAGE\n",
      "to identify serum microrna mirna as diagnostic and prognostic biomarkers for acute pancreatitis ap . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD identification of serum micrornas as diagnostic and prognostic biomarkers for acute pancreatitis . PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = identification of serum micrornas as diagnostic and prognostic biomarkers for acute pancreatitis . PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--20000.ckpt\n",
      "Iter 20200, accuracy = 0.1705, loss = 0.007543\n",
      "Iter 20400, accuracy = 0.1749, loss = 0.000831\n",
      "Iter 20600, accuracy = 0.1402, loss = 0.002937\n",
      "Iter 20800, accuracy = 0.1516, loss = 0.003575\n",
      "Iter 21000, accuracy = 0.1865, loss = 0.001118\n",
      "Iter 21000, VALIDATION -- accuracy = 0.1860, loss = 0.008150\n",
      "PASSAGE\n",
      "plants have evolved strategies and mechanisms to detect and respond to pathogen attack . different organs of the same plant may be subjected to different environments e . g . aboveground versus belowground and pathogens with different lifestyles . accordingly plants commonly need to tailor defense strategies in an organspecific manner . phytophthora infestans causal agent of potato late blight disease infects both aboveground foliage and belowground tubers . we examined the efficacy of transgene rb known for conferring foliar late blight resistance in defending against tuber late blight disease . our results indicate that the presence of the transgene has a positive yet only marginally significant effect on tuber disease resistance on average . however a significant association between transgene transcript levels and tuber resistance was established for specific transformed lines in an agedependent manner with higher transcript levels indicating enhanced tuber resistance . thus rb has potential to function in both foliage and tuber to impart late blight resistance . our data suggest that organspecific resistance might result directly from transcriptional regulation of the resistance gene itself . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD potato tuber blight resistance phenotypes correlate with rb transgene transcript levels in an agedependent manner . PAD PAD PAD\n",
      "Ground Truth = potato tuber blight resistance phenotypes correlate with rb transgene transcript levels in an agedependent manner . PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--21000.ckpt\n",
      "Iter 21200, accuracy = 0.1507, loss = 0.009089\n",
      "Iter 21400, accuracy = 0.1669, loss = 0.005568\n",
      "Iter 21600, accuracy = 0.1498, loss = 0.006746\n",
      "Iter 21800, accuracy = 0.1774, loss = 0.015708\n",
      "Iter 22000, accuracy = 0.1659, loss = 0.004027\n",
      "Iter 22000, VALIDATION -- accuracy = 0.1594, loss = 0.012069\n",
      "PASSAGE\n",
      "hematopoietic stem cell transplantation hsct can be curative for children with various malignant and nonmalignant conditions . access to suitable unrelated living donors UNK or unrelated cord blood ucb may be limited for certain ethnic backgrounds . we therefore determined the impact of ethnicity upon donor availability in a pediatric cohort referred for allogeneic hsct to a single canadian transplant center . among 252 eligible patients 58 23 . 0 had suitable family donors . of 161 patients with combined UNK and ucb searches 78 48 . 4 had a suitable UNK whereas 143 88 . 8 had suitable ucb . the probability of finding a suitable UNK differed significantly by ethnicity p0 . 007 . nonwhite patients were significantly less likely to have suitable UNK odds ratio or 0 . 35 95 confidence interval ci 0 . 170 . 69 p0 . 003 compared with white patients but were equally likely to have suitable ucb or 1 . 02 95 ci 0 . 362 . 89 p0 . 97 . although ethnic disparities exist in pediatric patients access to UNK for hsct they are narrowed by the availability of international ucb registries . these findings however also highlight the\n",
      "Model Pred = PAD impact of ethnicity on donor search results for children requiring stem cell transplantation . PAD PAD PAD PAD PAD\n",
      "Ground Truth = impact of ethnicity on donor search results for children requiring stem cell transplantation . PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--22000.ckpt\n",
      "Iter 22200, accuracy = 0.1618, loss = 0.004366\n",
      "Iter 22400, accuracy = 0.1832, loss = 0.006079\n",
      "Iter 22600, accuracy = 0.1833, loss = 0.000015\n",
      "Iter 22800, accuracy = 0.1903, loss = 0.003975\n",
      "Iter 23000, accuracy = 0.1648, loss = 0.006250\n",
      "Iter 23000, VALIDATION -- accuracy = 0.2047, loss = 0.006296\n",
      "PASSAGE\n",
      "methods for altering the sequence of endogenous drosophila melanogaster genes remain laborintensive . we have tested a relatively simple strategy that enables the introduction of engineered mutations in the vicinity of existing UNK . this method was used to generate useful alleles of the UNK gene which produces a noncoding rna involved in dosage compensation . the desired change was first introduced into a genomic clone of UNK and transgenic flies were generated that carry this sequence in a UNK . targeted transposition was then used to move the UNK into UNK . remobilization of the targeted insertion produced large numbers of offspring carrying chromosomes that had precisely introduced the engineered sequences into UNK . we postulate that this occurred by gap repair using the UNK on the sister chromatid as template . this strategy was used to introduce six ms2 loops into the UNK gene UNK enabling detection of UNK rna by a UNK fusion protein in embryos . the UNK remains under the control of the authentic promoter and within the correct genomic context features expected to contribute to normal UNK function . the ability to replace relatively large blocks of sequence suggests that this method will be\n",
      "Model Pred = PAD generation of a useful UNK allele by targeted gene conversion . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = generation of a useful UNK allele by targeted gene conversion . PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--23000.ckpt\n",
      "Iter 23200, accuracy = 0.1564, loss = 0.007057\n",
      "Iter 23400, accuracy = 0.1752, loss = 0.005997\n",
      "Iter 23600, accuracy = 0.1580, loss = 0.004560\n",
      "Iter 23800, accuracy = 0.1462, loss = 0.002416\n",
      "Iter 24000, accuracy = 0.1454, loss = 0.009476\n",
      "Iter 24000, VALIDATION -- accuracy = 0.1344, loss = 0.006805\n",
      "PASSAGE\n",
      "to report the profile of microbial keratitis occurring after corneal collagen crosslinking cxl in keratoconus patients . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD profile of microbial keratitis after corneal collagen crosslinking . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = profile of microbial keratitis after corneal collagen crosslinking . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--24000.ckpt\n",
      "Iter 24200, accuracy = 0.1759, loss = 0.001693\n",
      "Iter 24400, accuracy = 0.1836, loss = 0.004263\n",
      "Iter 24600, accuracy = 0.1576, loss = 0.004779\n",
      "Iter 24800, accuracy = 0.1470, loss = 0.007078\n",
      "Iter 25000, accuracy = 0.1491, loss = 0.000848\n",
      "Iter 25000, VALIDATION -- accuracy = 0.1729, loss = 0.006072\n",
      "PASSAGE\n",
      "to date there has been little research into the clinical outcomes of total hip arthroplasty tha for intracapsular neck of femur UNK fracture in the very elderly . 44 patients over UNK underwent tha for an intracapsular UNK fracture over a two year period . oxford hip scores were obtained from 37 patients with a mean score of 39 . 7 range UNK . UNK index scores were collected from 36 patients with a mean preoperative score of 5 . 9 and post operative score of 5 . 7 . tha in this population gives patients the best opportunity to return to premorbid function . when complications occur there is a catastrophic effect on independence . therefore it is important to select these patients appropriately and to optimise their medical condition perioperatively . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD clinical outcomes of total hip arthroplasty for fractured neck of femur in patients over 75 years . PAD PAD\n",
      "Ground Truth = clinical outcomes of total hip arthroplasty for fractured neck of femur in patients over 75 years . PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--25000.ckpt\n",
      "Iter 25200, accuracy = 0.1363, loss = 0.002488\n",
      "Iter 25400, accuracy = 0.1489, loss = 0.003197\n",
      "Iter 25600, accuracy = 0.1548, loss = 0.002612\n",
      "Iter 25800, accuracy = 0.1511, loss = 0.009778\n",
      "Iter 26000, accuracy = 0.1475, loss = 0.000239\n",
      "Iter 26000, VALIDATION -- accuracy = 0.1695, loss = 0.007535\n",
      "PASSAGE\n",
      "the recently solved crystal structure of the UNK cluster UNK UNK is examined using density functional theory dft . the UNK core of the cluster is intrinsically chiral by the arrangement of the au atoms . this is in contrast to the chirality of UNK gold clusters in which the protecting UNK units are arranged in chiral patterns on achiral cores . we interpret the electronic structure of the UNK cluster in terms of the UNK complex model . the UNK cluster cannot be interpreted as a dimer of UNK clusters which are magic . instead a UNK electron configuration of UNK UNK UNK UNK is found . the 2s band is strongly stabilized and the 1d states are nondegenerate with a large gap . ligand protection of the UNK core leads to a significant increase of the UNK and thus stabilization . we also tested a charge of ii which would give rise to an UNK UNK complex . our results indicate that the UNK cluster is indeed more stable . we also investigate the optical properties of the cluster . the experimental absorption spectrum is UNK by timedependent dft . prominent transitions are analyzed by timedependent densityfunctional perturbation\n",
      "Model Pred = PAD electronic structure and optical properties of the intrinsically chiral UNK UNK complex UNK . PAD PAD PAD PAD PAD\n",
      "Ground Truth = electronic structure and optical properties of the intrinsically chiral UNK UNK complex UNK . PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--26000.ckpt\n",
      "Iter 26200, accuracy = 0.1827, loss = 0.002855\n",
      "Iter 26400, accuracy = 0.1643, loss = 0.000218\n",
      "Iter 26600, accuracy = 0.1676, loss = 0.007563\n",
      "Iter 26800, accuracy = 0.1420, loss = 0.002908\n",
      "Iter 27000, accuracy = 0.1692, loss = 0.004691\n",
      "Iter 27000, VALIDATION -- accuracy = 0.1467, loss = 0.002129\n",
      "PASSAGE\n",
      "in japan membrane bioreactor mbrs have been installed in 17 smallscale municipal wastewater treatment plants wwtps in the past 8 years together with two recently installed mbrs for largerscale wwtps . in this study design and operating data were collected from 17 of them as part of a followup survey and aspects including system design biological treatment membrane operation problems and costs were UNK . because most of the mbrs were designed according to standardized guidance system configuration of the plants was similar UNK using the modified UNK mle process with membrane units submerged in aerobic tanks following a fine screen and flow equalization tank . this led to effluent quality with biochemical oxygen demand and tn of less than 3 . 5 and 7 . 4 mgl respectively for nine plants on an annual average basis . it was a common practice in extremely UNK plants to operate the membrane systems intermittently . frequency of recovery cleaning events was plantspecific mostly ranging from 1 to 5 UNK . cost evaluation revealed that specific construction costs for the smallscale mbrs were no more than for oxidation ditch plants . although specific energy consumption values tended to be high in the\n",
      "Model Pred = PAD design and operating experiences of fullscale municipal membrane bioreactors in japan . PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = design and operating experiences of fullscale municipal membrane bioreactors in japan . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--27000.ckpt\n",
      "Iter 27200, accuracy = 0.1626, loss = 0.005105\n",
      "Iter 27400, accuracy = 0.1524, loss = 0.000011\n",
      "Iter 27600, accuracy = 0.1823, loss = 0.000016\n",
      "Iter 27800, accuracy = 0.1728, loss = 0.002019\n",
      "Iter 28000, accuracy = 0.1837, loss = 0.004188\n",
      "Iter 28000, VALIDATION -- accuracy = 0.1779, loss = 0.005375\n",
      "PASSAGE\n",
      "a series of new watersoluble phenyl UNK conjugates containing hydrophilic UNK or UNK UNK moieties were synthesized via a UNK approach . these compounds have a broad spectrum of antitumor activity against a panel of human tumor cell lines . of these derivatives compound UNK effectively suppressed the growth of colon cancer hct116 prostate cancer pc3 and lung cancer h460 xenografts . the growth of hct116 xenografts was almost completely suppressed when cotreated with compound UNK and 5fluorouracil . furthermore compound UNK can induce dna crosslinking and cellcycle arrest at the g2m phase . early preclinical studies including pharmacokinetics in rats inhibition of the herg and 14 days of acute intravenous injection toxicity suggest that compound UNK is a promising candidate for further preclinical studies . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD design and synthesis of potent antitumor watersoluble phenyl UNK conjugates via a UNK approach . PAD PAD PAD PAD\n",
      "Ground Truth = design and synthesis of potent antitumor watersoluble phenyl UNK conjugates via a UNK approach . PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--28000.ckpt\n",
      "Iter 28200, accuracy = 0.1564, loss = 0.001819\n",
      "Iter 28400, accuracy = 0.1377, loss = 0.002342\n",
      "Iter 28600, accuracy = 0.1312, loss = 0.000187\n",
      "Iter 28800, accuracy = 0.1585, loss = 0.000299\n",
      "Iter 29000, accuracy = 0.1463, loss = 0.003452\n",
      "Iter 29000, VALIDATION -- accuracy = 0.1842, loss = 0.000213\n",
      "PASSAGE\n",
      "we aimed to evaluate the combination of the modified response evaluation criteria in solid tumors UNK and contrastenhanced ultrasonography ceus as a tool for the assessment of hepatocellular carcinoma treated with transarterial chemoembolization . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD UNK criteria and contrastenhanced us for the assessment of the response of hepatocellular carcinoma to transarterial chemoembolization . PAD\n",
      "Ground Truth = UNK criteria and contrastenhanced us for the assessment of the response of hepatocellular carcinoma to transarterial chemoembolization . PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--29000.ckpt\n",
      "Iter 29200, accuracy = 0.1609, loss = 0.000001\n",
      "Iter 29400, accuracy = 0.1200, loss = 0.007304\n",
      "Iter 29600, accuracy = 0.1610, loss = 0.004545\n",
      "Iter 29800, accuracy = 0.1649, loss = 0.002821\n",
      "Iter 30000, accuracy = 0.1416, loss = 0.000001\n",
      "Iter 30000, VALIDATION -- accuracy = 0.1783, loss = 0.002495\n",
      "PASSAGE\n",
      "studies of center of mass com motion are fundamental to understanding the dynamics of animal movement and have been carried out extensively for terrestrial and aerial locomotion . but despite a large amount of literature describing different body movement patterns in fishes analyses of how the center of mass moves during UNK propulsion are not available . these data would be valuable for understanding the dynamics of different body movement patterns and the effect of differing body shapes on locomotor force production . in the present study we analyzed the magnitude and frequency components of com motion in three dimensions x surge y sway z UNK in three fish species eel UNK UNK and UNK UNK swimming with four locomotor modes at three speeds using highspeed video and used an image crosscorrelation technique to estimate com motion thus enabling UNK and unrestrained locomotion . UNK swimming by eels shows reduced com surge oscillation magnitude relative to UNK swimming but not compared to UNK using a UNK locomotor style . UNK swimming UNK at 0 . 5 body UNK displays reduced com sway oscillation relative to swimming in a UNK style at higher speeds . oscillation frequency of the com in\n",
      "Model Pred = PAD center of mass motion in swimming fish effects of speed and locomotor mode during UNK propulsion . PAD PAD\n",
      "Ground Truth = center of mass motion in swimming fish effects of speed and locomotor mode during UNK propulsion . PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--30000.ckpt\n",
      "Iter 30200, accuracy = 0.1586, loss = 0.003375\n",
      "Iter 30400, accuracy = 0.1711, loss = 0.003805\n",
      "Iter 30600, accuracy = 0.1820, loss = 0.000629\n",
      "Iter 30800, accuracy = 0.1447, loss = 0.000118\n",
      "Iter 31000, accuracy = 0.1377, loss = 0.001822\n",
      "Iter 31000, VALIDATION -- accuracy = 0.1486, loss = 0.005095\n",
      "PASSAGE\n",
      "the aim of this study was to compare survival of patients with uterine sarcomas using the 1988 and 2008 international federation of gynecologists and obstetricians figo staging systems to determine if revised 2008 staging accurately predicts patient survival . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD figo staging for uterine sarcomas can the revised 2008 staging system predict survival outcome better PAD PAD PAD PAD\n",
      "Ground Truth = figo staging for uterine sarcomas can the revised 2008 staging system predict survival outcome better PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--31000.ckpt\n",
      "Iter 31200, accuracy = 0.1556, loss = 0.001298\n",
      "Iter 31400, accuracy = 0.1539, loss = 0.000584\n",
      "Iter 31600, accuracy = 0.1540, loss = 0.000007\n",
      "Iter 31800, accuracy = 0.1720, loss = 0.003267\n",
      "Iter 32000, accuracy = 0.1525, loss = 0.001199\n",
      "Iter 32000, VALIDATION -- accuracy = 0.1402, loss = 0.011505\n",
      "PASSAGE\n",
      "cardiothoracic surgeons and practitioners of cardiovascular medicine have a long history of humanitarian aid . although this is worthwhile at multiple levels and occasionally described in some detail few efforts have a proven algorithm with demonstrable outcomes that suggest effective educational methodology or clinical results approaching accepted standards in developed countries . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD a stepwise model for delivering medical humanitarian aid requiring complex interventions . PAD PAD PAD PAD PAD PAD PAD\n",
      "Ground Truth = a stepwise model for delivering medical humanitarian aid requiring complex interventions . PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--32000.ckpt\n",
      "Iter 32200, accuracy = 0.1508, loss = 0.000452\n",
      "Iter 32400, accuracy = 0.1453, loss = 0.004183\n",
      "Iter 32600, accuracy = 0.1681, loss = 0.000001\n",
      "Iter 32800, accuracy = 0.1671, loss = 0.000313\n",
      "Iter 33000, accuracy = 0.1639, loss = 0.000005\n",
      "Iter 33000, VALIDATION -- accuracy = 0.1516, loss = 0.003709\n",
      "PASSAGE\n",
      "dynamic mathematical modelling and stochastic simulation of UNK systems for the purpose of epidemiological analysis offer great opportunities for testing hypotheses especially when field experiments are impractical or when there is a need to evaluate multiple experimental scenarios . this combined with the ever increasing computer power available to researchers has contributed to the development of many mathematical models for epidemic simulations such as the individualbased model ibm . nevertheless few of these models undergo extensive validation and proper assessment of intrinsic variability . the ontario rabies model UNK will be used here to exemplify some advantages of appropriate model behaviour validation and to illustrate the use of a simple geometric procedure for testing directional bias in distributed stochastic dynamic model of spread of diseases . results were obtained through the comparison of 10 000 epizootics resulting from 100 epidemic simulations started using 100 distinct base populations . the analysis results demonstrated a significant directional bias in epidemic dispersion which prompted further verification of the model code and the identification of a coding error which was then corrected . subsequent testing of the corrected code showed that the directional bias could no longer be detected . these results illustrate the\n",
      "Model Pred = PAD a simple geometric validation approach to assess the basic behaviour of space and time distributed models of epidemic spread\n",
      "Ground Truth = a simple geometric validation approach to assess the basic behaviour of space and time distributed models of epidemic spread an\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--33000.ckpt\n",
      "Iter 33200, accuracy = 0.1715, loss = 0.000003\n",
      "Iter 33400, accuracy = 0.1916, loss = 0.000169\n",
      "Iter 33600, accuracy = 0.1668, loss = 0.001909\n",
      "Iter 33800, accuracy = 0.1378, loss = 0.000001\n",
      "Iter 34000, accuracy = 0.1595, loss = 0.007386\n",
      "Iter 34000, VALIDATION -- accuracy = 0.1606, loss = 0.008615\n",
      "PASSAGE\n",
      "there is a strong need to define the molecular changes in normal atrisk breast epithelium to identify biomarkers and new targets for breast cancer prevention and to develop a molecular signature for risk assessment . improved methods of breast epithelial sampling are needed to promote wholegenome molecular profiling increase ductal epithelial cell yield and reduce sample cell heterogeneity . PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "Model Pred = PAD an improved breast epithelial sampling method for molecular profiling and biomarker analysis in women at risk for breast cancer\n",
      "Ground Truth = an improved breast epithelial sampling method for molecular profiling and biomarker analysis in women at risk for breast cancer .\n",
      "\n",
      "Model saved in file: /home/ai2-rey/Documents/code/Nikhita/Text  Summarization /weights/seq2seq_weights_iter--34000.ckpt\n",
      "Iter 34200, accuracy = 0.1899, loss = 0.003785\n",
      "Iter 34400, accuracy = 0.1937, loss = 0.000349\n",
      "Iter 34600, accuracy = 0.1643, loss = 0.000012\n",
      "Iter 34800, accuracy = 0.1730, loss = 0.002702\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-43-b5c49ab21ea3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     21\u001b[0m                        batch_size: train_data_iter.batch_size}\n\u001b[1;32m     22\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m     \u001b[0msess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moptimizer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeed_dict\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtrain_feed_dict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mtrain_iter\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mdisplay_step\u001b[0m \u001b[0;34m==\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/ai2-rey/anaconda3/envs/tensorflow_env/lib/python3.5/site-packages/tensorflow/python/client/session.py\u001b[0m in \u001b[0;36mrun\u001b[0;34m(self, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[1;32m    764\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    765\u001b[0m       result = self._run(None, fetches, feed_dict, options_ptr,\n\u001b[0;32m--> 766\u001b[0;31m                          run_metadata_ptr)\n\u001b[0m\u001b[1;32m    767\u001b[0m       \u001b[0;32mif\u001b[0m \u001b[0mrun_metadata\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    768\u001b[0m         \u001b[0mproto_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtf_session\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTF_GetBuffer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrun_metadata_ptr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/ai2-rey/anaconda3/envs/tensorflow_env/lib/python3.5/site-packages/tensorflow/python/client/session.py\u001b[0m in \u001b[0;36m_run\u001b[0;34m(self, handle, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[1;32m    962\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mfinal_fetches\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mfinal_targets\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    963\u001b[0m       results = self._do_run(handle, final_targets, final_fetches,\n\u001b[0;32m--> 964\u001b[0;31m                              feed_dict_string, options, run_metadata)\n\u001b[0m\u001b[1;32m    965\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    966\u001b[0m       \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/ai2-rey/anaconda3/envs/tensorflow_env/lib/python3.5/site-packages/tensorflow/python/client/session.py\u001b[0m in \u001b[0;36m_do_run\u001b[0;34m(self, handle, target_list, fetch_list, feed_dict, options, run_metadata)\u001b[0m\n\u001b[1;32m   1012\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mhandle\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1013\u001b[0m       return self._do_call(_run_fn, self._session, feed_dict, fetch_list,\n\u001b[0;32m-> 1014\u001b[0;31m                            target_list, options, run_metadata)\n\u001b[0m\u001b[1;32m   1015\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1016\u001b[0m       return self._do_call(_prun_fn, self._session, handle, feed_dict,\n",
      "\u001b[0;32m/home/ai2-rey/anaconda3/envs/tensorflow_env/lib/python3.5/site-packages/tensorflow/python/client/session.py\u001b[0m in \u001b[0;36m_do_call\u001b[0;34m(self, fn, *args)\u001b[0m\n\u001b[1;32m   1019\u001b[0m   \u001b[0;32mdef\u001b[0m \u001b[0m_do_call\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1020\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1021\u001b[0;31m       \u001b[0;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1022\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mOpError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1023\u001b[0m       \u001b[0mmessage\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mas_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmessage\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/ai2-rey/anaconda3/envs/tensorflow_env/lib/python3.5/site-packages/tensorflow/python/client/session.py\u001b[0m in \u001b[0;36m_run_fn\u001b[0;34m(session, feed_dict, fetch_list, target_list, options, run_metadata)\u001b[0m\n\u001b[1;32m   1001\u001b[0m         return tf_session.TF_Run(session, options,\n\u001b[1;32m   1002\u001b[0m                                  \u001b[0mfeed_dict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfetch_list\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtarget_list\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1003\u001b[0;31m                                  status, run_metadata)\n\u001b[0m\u001b[1;32m   1004\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1005\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prun_fn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeed_dict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfetch_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for train_iter in range(train_iters):\n",
    "    train_iter +=1\n",
    "    train_batch, _ = train_data_iter.next_batch()\n",
    "    train_abstract, train_title, train_abs_len = zip(*train_batch)\n",
    "    if train_iter==1:\n",
    "        init_feed_dict = {passage:train_abstract,\n",
    "                          summary: train_title,\n",
    "                          dropout: 1.0,\n",
    "                          batch_size: train_data_iter.batch_size}\n",
    "        \n",
    "        init_loss, init_acc = tuple(sess.run([loss, accuracy], \n",
    "                                            feed_dict=init_feed_dict))\n",
    "        log_output = (\"Iter {}, accuracy = {:.4f}, loss = {:.6f}\").format(train_iter, init_acc, init_loss)\n",
    "        \n",
    "        print(log_output)\n",
    "        write_to_log(log_output, log_file)\n",
    "    \n",
    "    train_feed_dict = {passage:train_abstract, \n",
    "                       summary: train_title,\n",
    "                       dropout: keep_prob,\n",
    "                       batch_size: train_data_iter.batch_size}\n",
    "    \n",
    "    sess.run(optimizer, feed_dict=train_feed_dict)\n",
    "    \n",
    "    if train_iter % display_step ==0:\n",
    "        disp_feed_dict = {passage:train_abstract,\n",
    "                          summary: train_title,\n",
    "                          dropout: 1.0,\n",
    "                          batch_size: train_data_iter.batch_size}\n",
    "        \n",
    "        train_loss, train_acc = tuple(sess.run([loss, accuracy], \n",
    "                                               feed_dict=disp_feed_dict))\n",
    "        \n",
    "        log_output = (\"Iter {}, accuracy = {:.4f}, loss = {:.6f}\").format(train_iter, train_acc, train_loss)\n",
    "        print(log_output)\n",
    "        write_to_log(log_output, log_file)\n",
    "    \n",
    "    if test_data_iter !=None and train_iter% val_interval==0:\n",
    "        test_batch, _ = test_data_iter.next_batch()\n",
    "        test_abstract, test_title, test_abs_len = zip(*test_batch)\n",
    "        test_feed_dict = {passage:test_abstract,\n",
    "                          summary: test_title,\n",
    "                          dropout: 1.0,\n",
    "                          batch_size: test_data_iter.batch_size}\n",
    "        \n",
    "        test_loss, test_acc = tuple(sess.run([loss, accuracy], \n",
    "                                             feed_dict=test_feed_dict))\n",
    "        \n",
    "        log_output = (\"Iter {}, VALIDATION -- accuracy = {:.4f}, loss = {:.6f}\").format(train_iter, test_acc, test_loss)\n",
    "        print(log_output)\n",
    "        write_to_log(log_output, log_file)\n",
    "    \n",
    "    if deploy_data_iter !=None and train_iter % deploy_interval==0:\n",
    "        deploy_batch, _ = deploy_data_iter.next_batch()\n",
    "        deploy_abstract, deploy_title, dep_abs_lens = zip(*deploy_batch)\n",
    "        \n",
    "        model_pred = sess.run(generated_tokens, feed_dict={passage: deploy_abstract,\n",
    "                                                           summary: deploy_title,\n",
    "                                                           dropout: 1.0,\n",
    "                                                           batch_size: deploy_data_iter.batch_size})\n",
    "        \n",
    "        label_ids = deploy_title[0]\n",
    "        pred_ids = [x[0] for x in model_pred]\n",
    "        log_output = ''\n",
    "        if train.id2word_dict != None:\n",
    "            log_output = ('PASSAGE\\n{}\\nModel Pred = {}\\nGround Truth = {}\\n'\n",
    "                          .format(' '.join([train.id2word_dict[idx] for idx in deploy_abstract[0]]),\n",
    "                                  ' '.join([train.id2word_dict[idx] for idx in pred_ids]),\n",
    "                                  ' '.join([train.id2word_dict[idx] for idx in label_ids]))\n",
    "                         )\n",
    "        else:\n",
    "            log_output = ('Model Pred = {}, Ground Truth = {}'\n",
    "                          .format(pred_ids, label_ids)\n",
    "                         )\n",
    "        print(log_output)\n",
    "        write_to_log(log_output, log_file)\n",
    "    \n",
    "    if train_iter% save_weights_interval ==0:\n",
    "        weights_dir = model_dir + \"weights/seq2seq_weights_iter--{}.ckpt\".format(train_iter)\n",
    "        save_path = saver.save(sess,weights_dir)\n",
    "        save_string = \"Model saved in file: {}\".format(save_path)\n",
    "        print(save_string)\n",
    "        write_to_log(save_string, log_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda env:tensorflow_env]",
   "language": "python",
   "name": "conda-env-tensorflow_env-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
